File: 27. Remove Element/RemoveElement.cs
public class Solution {
    public int RemoveElement(int[] nums, int val) {
        
        int removedElements = 0;
        for (int i = 0; i < nums.Length; ++i) {

            if (nums[i] == val) {
                removedElements++;
            }
            else {
                nums[i-removedElements] = nums[i];
            }
        }
        return nums.Length-removedElements;
    }
}

File: 27. Remove Element/RemoveElement.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:

        removedElements = 0

        for index, num in enumerate(nums):
            if num == val:
                removedElements += 1
            else:
                nums[index-removedElements] = num
            
        return len(nums)-removedElements
Console
